src/Features/LanguageServer/Protocol/Handler/RequestContext.cs

```diff
@@ -116,6 +116,43 @@ namespace Microsoft.CodeAnalysis.LanguageServer.Handler;
                 throw new InvalidOperationException();
             }
```

Comment:
this method can just defer to this.TextDocument.  will remove redundancy/copying here.